controllers/restaurantOwners.controller.js

```diff
@@ -0,0 +1,65 @@
+const db = require('../db/index');
+
+var bcrypt = require('bcryptjs');
+var { check, validationResult } = require('express-validator');
+var passport = require('passport');
+
+exports.showHomePage = async (req, res, next) => {
+    try {
+        const topRestaurants = await db.any('SELECT rname, cuisine, raddress, round(AVG(rating)) AS avg FROM ReserveTimeSlots NATURAL JOIN OwnedRestaurants WHERE uname=$1 GROUP BY rname, cuisine, raddress ORDER BY AVG(rating) DESC, rname LIMIT 3', [req.params.uname]);
+        const allRestaurants = await db.any('SELECT * FROM OwnedRestaurants WHERE uname=$1', [req.params.uname]);
+        res.render('restaurantowners', {
+            title: 'Welcome ' + [req.params.uname] +'!',
+            topRestaurants: topRestaurants,
+            allRestaurants: allRestaurants
+        });
```

Comment:
This can be done in parallel, so can use `promise.all()` here instead!

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
Will attempt to solve this next time I want to die alr.